openlibrary/plugins/openlibrary/js/my-books/MyBooksDropper.js

```diff
@@ -0,0 +1,173 @@
+
+    /**
+     * Hydrates dropper contents and loads patron's lists.
+     */
+    initialize() {
+        super.initialize()
+
+        this.readingLogForms.initialize()
+        this.readingLists.initialize()
+
+        this.loadingAnimationId = this.initLoadingAnimation(this.dropper.querySelector('.loading-ellipsis'))
+    }
+
+    /**
+     * Creates loading animation for list loading indicator.
+     *
+     * @param {HTMLElement} loadingIndicator
+     * @returns {NodeJS.Timer}
+     */
+    initLoadingAnimation(loadingIndicator) {
+        let count = 0
+        const intervalId = setInterval(function() {
+            let ellipsis = ''
+            for (let i = 0; i < count % 4; ++i) {
+                ellipsis += '.'
+            }
+            loadingIndicator.innerText = ellipsis
+            ++count
+        }, 1500)
+
+        return intervalId
+    }
+
+    /**
+     * Replaces dropper loading indicator with the given
+     * partially rendered list affordances.
+     *
+     * @param {string} partialHtml
+     */
+    updateReadingLists(partialHtml) {
+        clearInterval(this.loadingAnimationId)
+        this.replaceLoadingIndicators(this.loadingIndicator, partialHtml)
+    }
+
+    /**
+     * Returns an array of seed keys associated with this dropper.
+     *
+     * If the seed identifies a book, there should be both an
+     * edition and work key in the results. Otherwise, the results
+     * array should only contain the primary seed key.
+     *
+     * @returns {Array<string>}
+     */
+    getSeedKeys() {
+        const results = [this.readingLists.seedKey]
+        if (this.readingLists.workKey) {
+            results.push(this.readingLists.workKey)
+        }
+        return results
+    }
+
+    /**
+     * Object returned by the list partials endpoint.
+     *
+     * @typedef {Object} ListPartials
+     * @property {string} dropper HTML string for dropdown list content
+     * @property {string} active HTML string for patron's active lists
+     */
+    /**
+     * Replaces list loading indicators with the given partially rendered HTML.
+     *
+     * @param {HTMLElement} dropperListsPlaceholder Loading indicator found inside of the dropdown content
+     * @param {ListPartials} partials
+     */
+    replaceLoadingIndicators(dropperListsPlaceholder, partialHTML) {
+        const dropperParent = dropperListsPlaceholder ? dropperListsPlaceholder.parentElement : null
+
+        if (dropperParent) {
+            removeChildren(dropperParent)
+            dropperParent.insertAdjacentHTML('afterbegin', partialHTML)
+
+            const anchors = this.dropper.querySelectorAll('.modify-list')
+            this.readingLists.initModifyListAffordances(anchors)
+        }
+    }
+
+    /**
+     * Closes dropper if opened; opens dropper if closed.
+     *
+     * If the dropper is open, the store is updated with
+     * a reference to the dropper and the seed identifier
+     * @override
+     */
+    toggleDropper() {
```

Comment:
I'm not sure what would be better for subclasses of `Dropper`:
1. Overriding `toggleDropper` if something needs to happen when a dropper's  state changes.  Or...
2. Including `onOpen` and `onClose` event hooks that could be overridden by `Dropper` subclasses if needed (the base methods would be empty).